apps/homepage/src/app/[lang]/contact/contactForm.tsx

```diff
@@ -0,0 +1,129 @@
+"use client";
+import { CustomInput } from "@1xdev/ui";
+import { CustomTextArea } from "@1xdev/ui";
+import styles from "./contactForm.module.scss";
+import { useLocalization } from "../../../lib/context/loc-context";
+import { UiValidator, UiValidatorErrors } from "@uireact/validator";
+import { useState } from "react";
+
+const validator = new UiValidator();
+
+export function ContactForm() {
+  const loc = useLocalization();
+
+  const schema = {
+    name: validator.field("text").present(loc.nameInputError),
+    email: validator.field("email").present(loc.emailInputError1),
+    phone: validator.field("numeric").present(loc.phoneInputError).length(10,12),
+    message: validator
+      .field("text")
+      .present(loc.messageInputError)
+      .length(0, 500),
+  };
+
+  const [contactInfo, setContactInfo] = useState({
+    name: "",
+    email: "",
+    phone: "",
+    message: "",
+  });
+
+  const [errors, setErrors] = useState<UiValidatorErrors>();
+
+  const handleChangeInputs = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    setContactInfo({
+      ...contactInfo,
+      [e.currentTarget.name]: e.currentTarget.value,
+    });
+  };
+
+  async function handleSubmit(e: React.FormEvent) {
```

Comment:
Let's convert this to a server action: https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations